### ISSUE TITLE
Validation perplexity explodes at first hours of training

### ISSUE THREAD

**Issue by ManiaCello** [pytorch/fairseq#572]
Hello! I'm trying to train Transformer big model with transformer_wmt_en_de_big configuration set for EN-RU language pair. It goes well for first 12500 steps, but then validation perplexity explodes from 118.51 to 5139.55 and training perplexity starts to monotonically increase. As I'm new to deep learning, I have few ideas what to do with it. Could you please give me a tip on what I can try to find the the cause of this problem?

I'm training on one Tesla K80 (Azure NC6 instance) with following training options:
`fairseq-train $NMT_DIR/ds/bin --max-tokens 1700 --arch transformer_wmt_en_de_big --max-epoch 5 \
            --update-freq 1 --optimizer adam --adam-betas '(0.9, 0.98)' \
            --learning-rate 0.0007 --clip-norm 0.0 \
            --lr-scheduler inverse_sqrt --warmup-init-lr 1e-07 --warmup-updates 4000 --min-lr 1e-09 \
             --criterion label_smoothed_cross_entropy --label-smoothing 0.1 --weight-decay 0.0\
             --save-dir $NMT_DIR/models \
             --no-progress-bar --log-format simple --log-interval 50 \
             --save-interval-updates  1000 --keep-interval-updates 5`

![valid_loss](https://user-images.githubusercontent.com/37402957/54249706-4643e400-4552-11e9-925f-e97aaf435d77.png)
![ppl](https://user-images.githubusercontent.com/37402957/54247306-75a22300-4549-11e9-92e5-c395b087b722.png)
![train_ppl](https://user-images.githubusercontent.com/37402957/54247474-2f00f880-454a-11e9-836c-729dfdac6cdf.png)
Training perplexity started growing since 12550 step.

Here is a log output when all that starts:

> | epoch 001:  11900 / 118626 loss=8.186, nll_loss=7.004, ppl=128.34, wps=987, ups=0.7, wpb=1467, bsz=51, num_updates=11900, lr=0.00040584, gnorm=2.570, clip=0%, oom=8.40266e-05, wall=17828, train_wall=14595
| epoch 001:  11950 / 118626 loss=8.183, nll_loss=7.001, ppl=128.12, wps=987, ups=0.7, wpb=1467, bsz=51, num_updates=11950, lr=0.00040499, gnorm=2.570, clip=0%, oom=8.3675e-05, wall=17896, train_wall=14657
| epoch 001 | valid on 'valid' subset | valid_loss 8.25533 | valid_nll_loss 6.88888 | valid_ppl 118.51 | num_updates 12000 | best 8.23211
| epoch 001:  12000 / 118626 loss=8.181, nll_loss=6.998, ppl=127.86, wps=987, ups=0.7, wpb=1466, bsz=51, num_updates=12000, lr=0.000404145, gnorm=2.567, clip=0%, oom=8.33264e-05, wall=17966, train_wall=14719
| epoch 001:  12050 / 118626 loss=8.178, nll_loss=6.995, ppl=127.55, wps=982, ups=0.7, wpb=1466, bsz=51, num_updates=12050, lr=0.000403306, gnorm=2.564, clip=0%, oom=8.29807e-05, wall=18147, train_wall=14779
| epoch 001:  12100 / 118626 loss=8.175, nll_loss=6.992, ppl=127.29, wps=982, ups=0.7, wpb=1466, bsz=51, num_updates=12100, lr=0.000402472, gnorm=2.562, clip=0%, oom=8.26378e-05, wall=18214, train_wall=14839
| epoch 001:  12150 / 118626 loss=8.172, nll_loss=6.989, ppl=127.03, wps=982, ups=0.7, wpb=1466, bsz=51, num_updates=12150, lr=0.000401643, gnorm=2.566, clip=0%, oom=8.22978e-05, wall=18282, train_wall=14900
| epoch 001:  12200 / 118626 loss=8.170, nll_loss=6.986, ppl=126.76, wps=983, ups=0.7, wpb=1466, bsz=51, num_updates=12200, lr=0.000400819, gnorm=2.568, clip=0%, oom=8.19605e-05, wall=18350, train_wall=14961

For a training set I joined 5 whole corpora (Yandex, TED, Wikipedia, Newscom11, EUbookshop) and also selected subsets from OpenSubtitles 2018 (3.5M) and from ParaCrawl (730.000). I tokenized with joint BPE encoding (with transliterated RU corpus)). I used this dataset already for training Transformer base model with OpenNMT system and didn't have any problems.
 
[training_log.txt](https://github.com/pytorch/fairseq/files/2959424/training_log.txt)

**Comment by huihuifan** [comment]
Are the parameters the same with fairseq and Open NMT?

**Comment by ManiaCello** [comment]
Hello, @huihuifan! No, I trained Transformer Base model (6 layers, 8 heads, 512 dimension size) at OpenNMT, and I tried to train Big model (transformer_wmt_en_de_big: 6 layers, 16 heads, 1024 dimension size) here.
Anyway, I decided to stop this experiment, because it's too hard to train such model with a single GPU (now I'm testing DynamicConv). So I'm closing the issue. Thank you for your reply!
